src/Features/Core/Portable/Completion/CompletionRules.cs

```diff
@@ -149,11 +149,23 @@ public virtual bool IsBetterFilterMatch(CompletionItem item1, CompletionItem ite
                 return false;
             }
 
+            // If they match on preselection and are from the same provider, 
+            // use their rules
+            if (item1.MatchPriority == item2.MatchPriority &&
+                item1.CompletionProvider == item2.CompletionProvider)
+            {
+                var result = item1.Rules.IsBetterPreselectedMatch(item1, item2, filterText);
```

Comment:
Curious: if we look at uses today of IsBetterPreselectedMatch, can we represent whatever that is in the MatchPriority behavior now, if we added another level or two?